sasmodels/sasview_model.py

```diff
@@ -697,42 +714,41 @@ def _calculate_Iq(self, qx, qy=None):
         #print("params", self.params)
         #print("values", values)
         #print("is_mag", is_magnetic)
+        if Fq:
+            result = calculator.Fq(call_details, values, cutoff=self.cutoff,
+                                   magnetic=is_magnetic,
+                                   effective_radius_type=effective_radius_type)
         result = calculator(call_details, values, cutoff=self.cutoff,
                             magnetic=is_magnetic)
+        lazy_results = getattr(calculator, 'results',
+                               lambda: collections.OrderedDict())
         #print("result", result)
-        self._intermediate_results = getattr(calculator, 'results', None)
+
         calculator.release()
         #self._model.release()
-        return result
 
-    def calculate_ER(self):
+        return result, lazy_results
+
+
+    def calculate_ER(self, mode=1):
         # type: () -> float
         """
         Calculate the effective radius for P(q)*S(q)
 
         :return: the value of the effective radius
         """
-        if self._model_info.ER is None:
-            return 1.0
-        else:
-            value, weight = self._dispersion_mesh()
-            fv = self._model_info.ER(*value)
-            #print(values[0].shape, weights.shape, fv.shape)
-            return np.sum(weight * fv) / np.sum(weight)
+        Fq = self._calculate_Iq([0.1], True, mode)
+        return Fq[2]
 
     def calculate_VR(self):
         # type: () -> float
         """
         Calculate the volf ratio for P(q)*S(q)
 
-        :return: the value of the volf ratio
+        :return: the value of the form:shell volume ratio
         """
-        if self._model_info.VR is None:
-            return 1.0
-        else:
-            value, weight = self._dispersion_mesh()
-            whole, part = self._model_info.VR(*value)
-            return np.sum(weight * part) / np.sum(weight * whole)
+        Fq = self._calculate_Iq([0.1], True, mode)
```

Comment:
mode is undefined in this scope. Do we need calculate_VR() at all?

Reply:
Don't know if sasview needs the results anywhere; probably not since this error hasn't popped up yet.  I've turned this into a NotImplementedError.